app/assets/javascripts/compute_resource.js

```diff
@@ -48,13 +49,16 @@ function testConnection(item) {
         $("#compute_resource_password").prop('disabled', false);
       }
       if (!/alert-danger/i.test(result)) {
-        notify("<p>" + __("Test connection was successful") + "</p>", 'success')
+        notify("<p>" + __("Test connection was successful") + "</p>", 'success');
+      } else {
+        $("#btn_datacenter").removeClass("btn-default").addClass("btn-danger");
```

Comment:
`compute resource#test connection`  renders `form` partial. if there are errors in the compute resource object, the `alert-danger` class is added (as you can see [here](https://github.com/theforeman/foreman/blob/develop/app/views/compute_resources/_form.html.erb#L3)). 

![alert_danger](https://cloud.githubusercontent.com/assets/11807069/14102492/9411861c-f5a3-11e5-98bd-f79b08551c4b.png)

Maybe `has-error` class is more likely to find.